src/event.c

```diff
@@ -333,29 +424,27 @@ void uag_event_unregister(ua_event_h *h)
  * @param fmt  Formatted arguments
  * @param ...  Variable arguments
  */
-void ua_event(struct ua *ua, enum ua_event ev, struct call *call,
-	      const char *fmt, ...)
+void ua_event(enum ua_event ev, const char *fmt, ...)
```

Comment:
Unsure if mixing formatting args and optional args is a good idea. It's very confusing.

Reply:
SInce this is `ua_event`, I would prefer to keep `struct ua *ua` param and add `event_param` (without `struct ua *ua;` in union).

Reply:
Yes.

And maybe we should not directly drop the current API.

I'll try a multi function solution for emitting events:
```c
/**
 * @deprecated
 */
void ua_event(struct ua *ua, enum ua_event ev, struct call *call, const char *fmt, ...);

void ua_event_app(enum ua_event ev, const char *fmt, ...);
void ua_event_ua(enum ua_event ev, struct ua *ua, const char *fmt, ...);
void ua_event_call(enum ua_event ev, struct call *call, const char *fmt, ...);
void ua_event_sip_msg(enum ua_event ev, struct sip_msg *msg, const char *fmt, ...);
```

This could be extended without breaking the API.